src/segmantic/seg/monai_unet.py

```diff
@@ -482,26 +472,26 @@ def train(
             else:
                 plt.show()
 
-            if output_dir:
+            if saver:
                 pred_labels = val_outputs.argmax(dim=1, keepdim=True)
                 saver.save_batch(pred_labels, val_data["image_meta_dict"])
 
 
 def predict(
-    model_file: str,
-    test_images: List[str],
-    test_labels: List[str] = None,
+    model_file: Path,
+    output_dir: Path,
+    test_images: List[Path],
+    test_labels: List[Path] = None,
```

Comment:
In the case of default "empty" lists as arguments, you did well adding None instead of an empty list. My approach in these cases is as follows

```python
  def foo( test_labels: Optional[List[Path]] = None):
     test_labels = test_labels or []. # it will be an empty list by default
     
```